Initial-Draft.md

```diff
@@ -0,0 +1,181 @@
+For example, let’s say you are writing a test for an input that is using the [AmountInput](https://storybook.affirm-dev.com/components-core/6.9.0/?path=/docs/components-forms-generic-numbers-amountinput--amount-input) reusable component. 
+If you were to place a data attribute on this component, the library might compile this attribute to something foreign at build time. 
+
+In this case you can try placing a data attribute on the parent element containing this component. 
+
+
+## Conditional Testing
+
+
+### Anti-pattern: Conditionals based on the existence of DOM elements
+
+While possible, this is not recommended because tests end up being non-deterministic, meaning different test runs may behave differently.  
+
+Example: Sometimes there is a Modal in front of the element you need to access for your test. So you may be thinking: “IF this modal exists, then close it, else just click my button…”. This is tempting since we deal with a lot of experiments, feature flags, country-based features, even cookies and local storage can affect what the user sees. 
+
+
+### Best Practices: 
+
+Remove the need to ever do conditional testing.
+
+Force your application to behave deterministically.
+
+According to Cypress docs: "The only way to do conditional testing on the DOM is if you are 100% sure that the state has "settled" and there is no possible way for it to change.... Unfortunately, it is not possible for you to use the DOM to do conditional testing. To do this would require you to know with 100% guarantee that your application has finished all asynchronous rendering and that there are no pending network requests, setTimeouts, intervals, postMessage, or async/await code."
+
+Example 1: 
+```js
+  cy.intercept('GET', shopSavingsEndpoints.eligible_instruments).as(
+    'getEligibleInstruments',
+  );
+
+  const getFirstInstrument = () => {
+    return cy.wait('@getEligibleInstruments').then(({ response }) => {
+      const { instruments }: { instruments: Instrument[] } = response?.body;
+      const activeInstrument = instruments.find(
+        (instrument: Instrument) => instrument.verification_status === 'active',
+      );
+
+      if (activeInstrument) {
+        cy.wrap(activeInstrument.title).as('instrumentTitle');
+        return cy.findByText(activeInstrument.title);
+      }
+      return false;
+    });
+  };
+```
+  
+The condional here is not based on the existence of a dom element, it is based on the stubbed out server resonse for eligible_instruments endpoint.
+
+
+Example 2:
+```js
+  if (Cypress.browser.name !== 'chrome') {
+    console.log('SKIPPING TEST. ONLY RUNS FOR CHROME.');
+    return;
+  }
+```
+
+While .cy is async, Cypress.browser is not. This method returns the browser object with properties like `name`. This is safe to do. 
+
+See:
```

Comment:
I'm not sure if Cypress has documentation for dealing with third party components, and if we use third party components at all. Based on my experience, some third party components have weird implementations which makes it hard to work with cypress (e.g. the dom element will get mounted and dismounted constantly which makes it hard for cypress to grab...)